metagraph/src/cli/assemble.cpp

```diff
@@ -107,6 +108,40 @@ mask_graph(const AnnotatedDBG &anno_graph, Config *config) {
 }
 
 
+std::string sequence_to_gfa_path(const std::string &seq,
+                                 const size_t seq_id,
+                                 const std::shared_ptr<DeBruijnGraph> &graph,
+                                 const tsl::ordered_set<uint64_t> &is_unitig_end_node,
+                                 const Config *config) {
+    std::vector<uint64_t> path_nodes;
+    graph->map_to_nodes_sequentially(seq, [&](uint64_t node) {
+        path_nodes.push_back(node);
+    });
+
+    //    The first node in the path must be a unitig end node.
+    assert(is_unitig_end_node.count(path_nodes[0]));
```

Comment:
Are the PR tests running the integration tests in a different way? Maybe different flags? In the git checks this assert is failing, but while running locally everything is good (`python3 /home/radu/projects/projects2014-metagenome/metagraph/integration_tests/main.py --test_filter "*assemble_gfa*"`)

Reply:
Hm, the CI should run the same command...

Reply:
It is fixed now. I don't understand why yet, but only the github checks in Debug environment were failing. By building the project locally with "-DCMAKE_BUILD_TYPE=Debug" I was able to find and fix the issue. Thanks for reviewing my PRs!